### DISCUSSION TITLE
Formatter to use with Nebula?

### DISCUSSION THREAD

**Discussion by laeubi** [EclipseNebula/nebula#604] [Q&A]
I just noticed that almost all changes in my IDE generate a large set of unrelated formatting changes as every widget seem to use its own or even a mixed style of code formatting.

As I don't found anything I wonder if / what to use when formatting source-code, I would actually be happy with any Eclipse-Buit-In e.g. Java Convention, Eclipse (2.1) or alternatively a custom one located somewhere in the repository.

FYI @lcaron @wimjongman

**Comment by wimjongman** [comment]
Consider setting 'only format changed lines' in the preferences.

I think that we use the default eclipse settings.

@lcaron?

Best regards,

Wim Jongman
________________________________
From: Christoph Läubrich ***@***.***>
Sent: Sunday, July 21, 2024 1:54:54 PM
To: eclipse/nebula ***@***.***>
Cc: Wim Jongman ***@***.***>; Mention ***@***.***>
Subject: [eclipse/nebula] Formatter to use with Nebula? (Discussion #604)


I just noticed that almost all changes in my IDE generate a large set of unrelated formatting changes as every widget seem to use its own or even a mixed style of code formatting.

As I don't found anything I wonder if / what to use when formatting source-code, I would actually be happy with any Eclipse-Buit-In e.g. Java Convention, Eclipse (2.1) or alternatively a custom one located somewhere in the repository.

FYI @lcaron<https://github.com/lcaron> @wimjongman<https://github.com/wimjongman>

**Reply by laeubi** [reply]
> Consider setting 'only format changed lines' in the preferences.

Yes that's what I'm currently doing, but as it is not the default setting I sometimes miss it ...
Also reading the code is often hard with a mixed style for example this file here:

https://github.com/eclipse/nebula/blob/31f40e4e56e006f54e87eae14b23c36dbd79f09b/widgets/formattedtext/org.eclipse.nebula.widgets.formattedtext/src/org/eclipse/nebula/widgets/formattedtext/NumberFormatter.java#L396-L412

so my idea would be to reformat such files once before apply a change to them so they are consistently formatted first.

**Reply by wimjongman** [reply]
Yes. Please first commit, merge and push the reformatting and then the actual change. This can be done in the same issue.

**Reply by laeubi** [reply]
> Yes. Please first commit, merge and push the reformatting

That's why I like what exactly to use for reformatting :-)

**Reply by wimjongman** [reply]
Just use the default settings in Eclipse.

**Reply by wimjongman** [reply]
If you wish, you can store the formatting preferences inside the project.

**Reply by lcaron** [reply]
Hi

I do not think that we have a rule of formatting. In my opinion we should use the default Eclipse one.

Laurent
